ports/mimxrt10xx/board_flash.c

```diff
@@ -55,13 +55,15 @@ void board_flash_init(void)
 {
   flexspi_nor_flash_init(FLEXSPI_INSTANCE, (flexspi_nor_config_t*) &qspiflash_config);
 
-  // Check for FCFB and copy bootloader to flash if not present
-  if ( *(uint32_t*) FCFB_START_ADDRESS != FLEXSPI_CFG_BLK_TAG )
+  // TinyUF2 will copy its image to flash if  Boot Mode is '01' i.e Serial Download Mode (BootRom)
+  // Normally it is done once by SDPHost or used to recover an corrupted boards
+  uint32_t const boot_mode = (SRC->SBMR2 & SRC_SBMR2_BMOD_MASK) >> SRC_SBMR2_BMOD_SHIFT;
+  if (boot_mode == 1)
```

Comment:
Should we check the revision to prevent accidentally downgrading?

Reply:
I thought of this, however there is actually no need to check against SDP. SDP is either production mode or the desperate mode. When used it should just overwrite whatever on the flash.

Reply:
That is safest for de-bricking.  There is an outside chance that they grab an older .bin than was previously loaded on the board but if they switched to serial download mode it is probably safer to let them load it anyway.

Reply:
yup, for version stepping, it is doable in much easier way with drag & drop `update-tinyuf2.uf2` is really the de-brick move when user accidentally corrupt flash config and/or bootloader